File: depyler_tests/test_008_trait_object_class_field.py
import sys

class Logger:
    def __init__(self, filename: str = None):
        if filename:
            self.output = open(filename, "w")
        else:
            self.output = sys.stdout

    def log(self, msg: str):
        self.output.write(f"[LOG] {msg}\n")

def main():
    logger = Logger()
    logger.log("test message")

File: depyler_tests/test_008_trait_object_class_field.rs
use std as sys;
#[derive(Debug, Clone)]
pub struct Logger {}
impl Logger {
    pub fn new(filename: String) -> Self {
        Self {}
    }
    pub fn log(&self, msg: String) {
        self.output.write(format!("[LOG] {}\n", msg));
    }
}
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn main() {
    let logger = Logger::new();
    logger.log("test message".to_string());
}
